aws/table_aws_ec2_launch_template.go

```diff
@@ -0,0 +1,189 @@
+		List: &plugin.ListConfig{
+			IgnoreConfig: &plugin.IgnoreConfig{
+				ShouldIgnoreErrorFunc: shouldIgnoreErrors([]string{"InvalidLaunchTemplateName.NotFoundException", "InvalidLaunchTemplateId.NotFound", "InvalidLaunchTemplateId.Malformed"}),
+			},
+			Hydrate: listEc2LaunchTemplates,
+			KeyColumns: []*plugin.KeyColumn{
+				{
+					Name:    "launch_template_name",
+					Require: plugin.Optional,
+				},
+				{
+					Name:    "launch_template_id",
+					Require: plugin.Optional,
+				},
+			},
+		},
+		GetMatrixItemFunc: SupportedRegionMatrix(ec2v1.EndpointsID),
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "launch_template_name",
+				Description: "The name of the launch template.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "launch_template_id",
+				Description: "The ID of the launch template.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "create_time",
+				Description: "The time launch template was created.",
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
+				Name:        "created_by",
+				Description: "The principal that created the launch template.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "default_version_number",
+				Description: "The version number of the default version of the launch template.",
+				Type:        proto.ColumnType_INT,
+			},
+			{
+				Name:        "latest_version_number",
+				Description: "The name of the Application-Layer Protocol Negotiation (ALPN) policy.",
+				Type:        proto.ColumnType_INT,
+			},
+			{
+				Name:        "tags_src",
+				Description: "The tags for the launch template.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Tags"),
+			},
+
+			// Standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("LaunchTemplateName"),
+			},
+			{
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.From(launchTemplateTurbotTags),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     launchTemplateAkas,
+				Transform:   transform.FromValue(),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listEc2LaunchTemplates(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// Create Session
+	svc, err := EC2Client(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_ec2_launch_template.listEc2LaunchTemplates", "connection_error", err)
+		return nil, err
+	}
+
+	// Limiting the results
+	maxLimit := int32(200)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxLimit {
+			if limit < 1 {
+				maxLimit = 1
```

Comment:
I don't think we need this check since the Steampipe plugin SDK handles it automatically.